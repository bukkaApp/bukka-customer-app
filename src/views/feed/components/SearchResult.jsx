/* eslint-disable react/prop-types */
import React, { useEffect, useState } from 'react';

import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter, useHistory } from 'react-router-dom';
import InfiniteScroll from 'react-infinite-scroller';
import shortId from 'shortid';

import { useLocationContext } from 'src/context/LocationContext';
import BukkaCard from 'src/components/Carousel/BukkaCard';
import Row from 'src/components/grid/Row';
import Container from 'src/components/container/Container';
import DeliveryOrPickupNav from 'src/components/common-navs/DeliveryOrPickupNav';
import fetchBukkaMenuAction from 'src/redux/fetchBukkaMenuAction';
import getBukkasRelatedToSingleCuisines from '../actionCreators/getBukkasRelatedToSingleCuisines';
import getMoreBukkasRelatedToSingleCuisines from '../actionCreators/getMoreBukkasRelatedToSingleCuisines';



import fetchBukkas from '../actionCreators/fetchBukkas';
import IntroSection from '../common/IntroSection';
import ExploreSection from '../common/ExploreSection';

// TODO: Don't  display time if bukkas are not avaailable or they have closed

import './searchresult.scss';

const Headline = ({ title, views }) => (
  <Container>
    <div className="headline">
      <blockquote>
        <h2 className="headline-h2">
          {title}
        </h2>
      </blockquote>
      {views > 0 && (
        <a className="headline-link" href="/" rel="nofollow">
          <span className="d-none pr-3 d-sm-inline-flex">
            {views} {views > 1 ? 'Results' : 'Result'}
          </span>
        </a>
      )}
    </div>
  </Container>
);

const FoodSection = ({
  search,
  fetchNearbyBukkas,
  status: { error },
  location: match,
  cuisineItems,
  errorMessage,
  currentPage,
  cuisineToDisplay: { name },
  getMoreBukkasRelatedToCuisines,
  getBukkasRelatedToCuisines,
  fetchBukkaMenu,
}) => {
  const { push } = useHistory();
  const { coordinates } = useLocationContext();
  const [searchQuery, setSearchQuery] = useState({
    by: '',
    value: ''
  });
  const [searchResults, setSearchResults] = useState([]);

  useEffect(() => {
    console.log('match', match);
    let suscribed = true;//eslint-disable-line
    getBukkasRelatedToCuisines('american', coordinates);
    return () => {
      suscribed = false;
    };
  }, [coordinates]);

  useEffect(() => {
    fetchNearbyBukkas(coordinates, 1, 100, searchQuery.by, searchQuery.value);
  }, [coordinates, searchQuery]);

  useEffect(() => {
    if (cuisineItems.length > 0 && searchQuery.value.length > 0) {
      const newSearchResult = cuisineItems.filter(bukka =>
        bukka[searchQuery.by].includes(searchQuery.value)
      );
      setSearchResults(newSearchResult);
    }
  }, [cuisineItems, searchQuery]);

  useEffect(() => {
    setSearchQuery({ by: 'name', value: search });
    push(`/search?by=name&value=${search}`);
  }, [search]);

  const handleChange = () => {};

  return (
    <div className="search-result-scene container-fluid p-0">
      <div>
        <IntroSection handleChange={handleChange} push={push} />
        <ExploreSection classNames="pt-5">
          <div className="mt-130 mb-4">
            {searchResults.length === 0 && <Headline title={searchQuery.value || 'Search'} />}
            {searchResults.length > 0 && (
              <Headline
                title={`${searchQuery.value}`}
                views={searchResults}
              />
            )}
            <Container classNames="px-0">
              {searchResults.length > 0 ? (
                <div className="col-md-3 col-lg-2">
                  <DeliveryOrPickupNav />
                </div>
              ) : (
                <h6 className="pl-3">EXPLORE TOP CUISINES</h6>
              )}
            </Container>
            <>
              {searchResults.length === 0 && searchQuery.value.length > 0 && (
                <h2 className="text-center" style={{ marginTop: '50px' }}>
                  Your search returned no results
                </h2>
              )}
            </>
            <Container>
              {searchResults.length > 0 && (
                <Row classNames="pb-4">
                  {searchResults.map(bukka => (
                    <BukkaCard
                      key={shortId.generate()}
                      imageUrl={bukka.imageUrl}
                      mealName={bukka.name}
                      deliveryPrice={bukka.deliveryPrice}
                      deliveryTime={bukka.deliveryTime}
                      rating={bukka.rating}
                      imageHeight="img-height"
                      classNames="col-xl-4 col-md-6 col-sm-12"
                      dataTarget="#bukkaAddToCart"
                      dataToggle="modal"
                    />
                  ))}
                </Row>
              )}
              {/* show categories if there is no search */}
              <InfiniteScroll
                loadMore={() =>
                  getMoreBukkasRelatedToCuisines(
                    'american', coordinates, Number(currentPage) + 1)}
                hasMore={errorMessage === ''}
                loader={
                  <div className="loader text-center" key={0}>
                    <div className="spinner-grow custom-text-color" role="status">
                      <span className="sr-only">Loading...</span>
                    </div>
                  </div>
                }
                useWindow
                initialLoad={false}
              >
                <Row classNames="pb-4">
                  {cuisineItems.map(bukka => (
                    <BukkaCard
                      key={shortId.generate()}
                      imageUrl={bukka.imageUrl}
                      mealName={bukka.name}
                      delivery={false}
                      handleClick={() => fetchBukkaMenu(`/bukka/${bukka.slug}`)}
                      deliveryPrice={bukka.deliveryPrice}
                      deliveryTime={bukka.deliveryTime}
                      rating={bukka.rating}
                      tags={bukka.placeGroup}
                      imageHeight="img-fluid"
                      classNames="col-xl-4 col-md-6 col-sm-12"
                      href={`/bukka/${bukka.slug}`}
                    />
                  ))}
                </Row>
              </InfiniteScroll>
            </Container>
          </div>
        </ExploreSection>
      </div>
    </div>
  );
};

const mapStateToProps = ({
  deliveryModeReducer: { mode },
  searchAnythingReducer: { search },
  bukkasReducer: { fetchedBukkas, status },
  cuisineReducer: {
    cuisineItems, errorMessage,
    currentPage,
    cuisineToDisplay: { name },
  },
}) => ({
  search,
  fetchedBukkas,
  status,
  mode,
  cuisineItems,
  errorMessage,
  currentPage,
  cuisineToDisplay: { name },
});

export default connect(
  mapStateToProps,
  { fetchNearbyBukkas: fetchBukkas,
    fetchBukkaMenu: fetchBukkaMenuAction,
    getBukkasRelatedToCuisines: getBukkasRelatedToSingleCuisines,
    getMoreBukkasRelatedToCuisines: getMoreBukkasRelatedToSingleCuisines, }
)(withRouter(FoodSection));

FoodSection.propTypes = {
  search: PropTypes.string.isRequired,
  fetchedBukkas: PropTypes.shape({
    nearbyBukkas: PropTypes.arrayOf(PropTypes.shape({}))
  }).isRequired,
  fetchNearbyBukkas: PropTypes.func.isRequired,
  status: PropTypes.objectOf(PropTypes.bool).isRequired
};

Headline.defaultProps = {
  views: 0
};

Headline.propTypes = {
  views: PropTypes.number,
  title: PropTypes.string.isRequired
};
