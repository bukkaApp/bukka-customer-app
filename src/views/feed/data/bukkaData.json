[
  {
    "deliveryTime": "40-50 min",
    "title": "burger",
    "deliveryPrice": 300,
    "imageUrl": "https://res.cloudinary.com/deqt3envc/image/upload/v1556741830/samples/Bukka-Images/carousel-burger-img.png",
    "rating": "popular",
    "subHeading": "order now",
    "_id": "02994249",
    "position": {
      "top": true
    }
  },
  {
    "deliveryTime": "40-50 min",
    "title": "ice dougnut",
    "deliveryPrice": 300,
    "imageUrl": "https://res.cloudinary.com/deqt3envc/image/upload/v1556741830/samples/Bukka-Images/carousel-ice-dougnut.png",
    "rating": "popular",
    "subHeading": "order now",
    "_id": "0234594249",
    "position": {
      "top": true
    }
  },
  {
    "deliveryTime": "40-50 min",
    "title": "alcholic drinks",
    "deliveryPrice": 300,
    "imageUrl": "https://res.cloudinary.com/deqt3envc/image/upload/v1556741830/samples/Bukka-Images/carousel-alcholic-drinks.png",
    "rating": "popular",
    "heading": "Get Drinks",
    "_id": "029992034249",
    "subHeading": "ALCOHOL DELIVERED",
    "position": { "top": true }
  },
  {
    "deliveryTime": "40-50 min",
    "title": "fruits",
    "deliveryPrice": 300,
    "imageUrl": "https://res.cloudinary.com/deqt3envc/image/upload/v1556741830/samples/Bukka-Images/carousel-fruits.png",
    "rating": "popular",
    "heading": "Get Fresh",
    "_id": "029273561249",
    "subHeading": "Groceries Delivered",
    "position": { "top": true }
  },
  {
    "deliveryTime": "40-50 min",
    "title": "free burger",
    "deliveryPrice": 300,
    "_id": "0299425723234",
    "imageUrl": "https://res.cloudinary.com/deqt3envc/image/upload/v1556741830/samples/Bukka-Images/carousel-free-burger.png",
    "rating": "popular",
    "heading": "Get All Orders Delivered Free",
    "subHeading": "TRY UNLIMITED",
    "textPositionBottom": true
  }
]
