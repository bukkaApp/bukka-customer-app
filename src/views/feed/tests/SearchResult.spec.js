import React from 'react';

import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';

import freeDelivery from '../data/free-delivery.json';
import bukkaData from '../data/search.json';
import SearchResult from 'src/components/SearchResult';

const initialState = {
  navbarAuthReducer: { type: 'Sign In', },
  deliveryModeReducer: { mode: 'delivery', },
  searchAnythingReducer: { search: '' },
  locationsPredictionReducer: { predictions: [3.7474, 3.4848] },
  authenticationReducer: {
    status: {
      authenticated: false,
      error: false
    }
  },
  bukkasReducer: {
    fetchedBukkas: {
      nearbyBukkas: [],
      message: ''
    },
    status: {
      fetchedBukkas: false,
      error: true,
    }
  },
  selectedLocationReducer: { coordinates: [3.7474, 3.4848] }
};

const store = mockStore(initialState);

describe.skip('SearchResult component', () => {
  const props = {
    signOut: jest.fn(),
    setDeliveryModeAction: jest.fn(),
    push: jest.fn(),
    coordinates: [3.7474, 3.4848],
    fetchedBukkas: { nearbyBukkas: [] },
    fetchNearbyBukkas: jest.fn(),
    status: { error: false },
    cuisineReducer: {
      cuisineItems: [],
      errorMessage: '',
      currentPage: 1,
      cuisineToDisplay: { name: '' },
    },
  };
  const { container, rerender } = render(
    <Provider store={store}>
      <MemoryRouter>
        <SearchResult {...props} />
      </MemoryRouter>
    </Provider>
  );

  it.skip('renders properly', () => {
    expect(container).toMatchSnapshot();
  });
  it.skip('renders properly', () => {
    const nearbyBukkas = [...freeDelivery, ...bukkaData];
    const newState = {
      ...initialState,
      searchAnythingReducer: { search: 'a' },
      bukkasReducer: {
        ...initialState.bukkasReducer,
        fetchedBukkas: {
          ...initialState.bukkasReducer.fetchedBukkas,
          nearbyBukkas,
          message: ''
        },
        status: {
          ...initialState.bukkasReducer.status,
          fetchedBukkas: true,
          error: false
        }
      }
    };
    const newProps = {
      ...props,
      fetchedBukkas: { nearbyBukkas },
    };
    const newStore = mockStore(newState);
    rerender(
      <Provider store={newStore}>
        <MemoryRouter>
          <SearchResult {...newProps} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
