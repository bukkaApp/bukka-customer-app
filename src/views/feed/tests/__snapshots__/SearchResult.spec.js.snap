// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResult component renders properly 1`] = `
<div>
  <div>
    <div
      class="container"
    >
      <div
        aria-hidden="true"
        aria-labelledby="exampleModalLabel"
        class="modal fade"
        id="authModal"
        role="dialog"
        tabindex="-1"
      >
        <div
          class="modal-dialog modal-dialog-centered auth-modal"
          role="document"
        >
          <div
            class="modal-content"
          >
            <div
              class="modal-header"
            />
            <div
              class="modal-body"
            >
              <div
                class="dismiss-modal close"
                data-dismiss="modal"
              >
                <svg
                  fill="currentColor"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                >
                  <defs>
                    <path
                      d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                      id="icon-close_svg__a"
                    />
                  </defs>
                  <use
                    transform="translate(4 4)"
                    xlink:href="#icon-close_svg__a"
                  />
                </svg>
              </div>
              <div
                class="bg-color auth-page"
              >
                <div
                  class=""
                >
                  <div
                    class="auth-wrapper pt-5"
                  >
                    <h2
                      class="text-center pt-4 mb-0 auth-title"
                    >
                      Log In
                    </h2>
                    <form
                      class="pb-3 form-auth"
                      name="form"
                    >
                      <div
                        class="form-title mb-4"
                      >
                        <span>
                          Type your email
                        </span>
                      </div>
                      <div
                        class="input-group-wrapper mt-9"
                      >
                        <div
                          class="input-slide text-center "
                        >
                          <div
                            class="form-group padding"
                          >
                            <input
                              accept=""
                              autocomplete=""
                              class="default-input"
                              name="email"
                              placeholder="Email"
                              type="email"
                              value=""
                            />
                            <div
                              class="text-danger help-block"
                            />
                          </div>
                          <div
                            class="form-group padding"
                          >
                            <input
                              accept=""
                              autocomplete=""
                              class="default-input"
                              name="password"
                              placeholder="Password"
                              type="password"
                              value=""
                            />
                            <div
                              class="text-danger help-block"
                            />
                          </div>
                        </div>
                        <div
                          aria-pressed="false"
                          class="col-lg-12 padding terms text-center"
                          role="button"
                          tabindex="0"
                        >
                          <p
                            class="link text-muted"
                          >
                            Forgot Password?
                          </p>
                        </div>
                      </div>
                      <div
                        class="padding"
                      >
                        <button
                          class="button col-md-12 primary-button mt-5"
                          data-dismiss=""
                          data-target=""
                          data-testid="button"
                          data-toggle=""
                          id="submit"
                          type="submit"
                        >
                          Log In
                        </button>
                      </div>
                      <div
                        class="divider-box mt-2 mb-2"
                      >
                        <div
                          class="divider"
                        >
                          <span>
                            or
                          </span>
                        </div>
                      </div>
                      <div
                        class="padding"
                      >
                        <button
                          class="facebk-btn col-md-12"
                          data-dismiss=""
                          data-target=""
                          data-testid="button"
                          data-toggle=""
                          id="2jjaYY3qn"
                          type="button"
                        >
                          Facebook
                        </button>
                      </div>
                      <div
                        class="form-options padding"
                      >
                        <p>
                          New to Bukka?
                        </p>
                        <button
                          class="btn-link auth-footer-btn"
                          data-dismiss=""
                          data-target=""
                          data-testid="button"
                          data-toggle=""
                          id="/signup"
                          type="button"
                        >
                          SIGN UP
                        </button>
                      </div>
                    </form>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class=""
    >
      <nav
        class="container navbar navbar-light"
      >
        <div>
          <a
            class="bukka-brand"
            href="/"
          >
            <div
              class="bukka-logo"
            >
              <img
                alt=""
                src="https://res.cloudinary.com/digitzs/image/upload/v1550480450/Bukka%20App/bukka-logo.svg"
              />
            </div>
          </a>
        </div>
        <div
          class="form-inline"
        >
          <button
            class="small-outline-button"
            data-dismiss=""
            data-target="#authModal"
            data-testid="button"
            data-toggle="modal"
            id="/login"
            type="button"
          >
            sign in
          </button>
          <button
            class="small-button"
            data-dismiss=""
            data-target="#authModal"
            data-testid="button"
            data-toggle="modal"
            id="/signup"
            type="button"
          >
            sign up
          </button>
        </div>
      </nav>
    </div>
    <div
      class="bg-color not-available-page"
    >
      <div
        class="container "
      >
        <h1
          class="text-title-unavailable"
        >
          Mybuka Not
          <br />
          Available In Your
          <br />
          Location
          <br />
        </h1>
      </div>
    </div>
  </div>
</div>
`;

exports[`SearchResult component renders properly 2`] = `
<div>
  <div
    class="search-result-scene container-fluid p-0"
  >
    <div>
      <div
        class="intro-navbar-area"
      >
        <div
          class="container"
        >
          <div
            aria-hidden="true"
            aria-labelledby="exampleModalLabel"
            class="modal fade"
            id="authModal"
            role="dialog"
            tabindex="-1"
          >
            <div
              class="modal-dialog modal-dialog-centered auth-modal"
              role="document"
            >
              <div
                class="modal-content"
              >
                <div
                  class="modal-header"
                />
                <div
                  class="modal-body"
                >
                  <div
                    class="dismiss-modal close"
                    data-dismiss="modal"
                  >
                    <svg
                      fill="currentColor"
                      height="24"
                      viewBox="0 0 24 24"
                      width="24"
                    >
                      <defs>
                        <path
                          d="M0 1.5L1.5 0 8 6.5 14.5 0 16 1.5 9.5 8l6.5 6.5-1.5 1.5L8 9.5 1.5 16 0 14.5 6.5 8z"
                          id="icon-close_svg__a"
                        />
                      </defs>
                      <use
                        transform="translate(4 4)"
                        xlink:href="#icon-close_svg__a"
                      />
                    </svg>
                  </div>
                  <div
                    class="bg-color auth-page"
                  >
                    <div
                      class=""
                    >
                      <div
                        class="auth-wrapper pt-5"
                      >
                        <h2
                          class="text-center pt-4 mb-0 auth-title"
                        >
                          Log In
                        </h2>
                        <form
                          class="pb-3 form-auth"
                          name="form"
                        >
                          <div
                            class="form-title mb-4"
                          >
                            <span>
                              Type your email
                            </span>
                          </div>
                          <div
                            class="input-group-wrapper mt-9"
                          >
                            <div
                              class="input-slide text-center "
                            >
                              <div
                                class="form-group padding"
                              >
                                <input
                                  accept=""
                                  autocomplete=""
                                  class="default-input"
                                  name="email"
                                  placeholder="Email"
                                  type="email"
                                  value=""
                                />
                                <div
                                  class="text-danger help-block"
                                />
                              </div>
                              <div
                                class="form-group padding"
                              >
                                <input
                                  accept=""
                                  autocomplete=""
                                  class="default-input"
                                  name="password"
                                  placeholder="Password"
                                  type="password"
                                  value=""
                                />
                                <div
                                  class="text-danger help-block"
                                />
                              </div>
                            </div>
                            <div
                              aria-pressed="false"
                              class="col-lg-12 padding terms text-center"
                              role="button"
                              tabindex="0"
                            >
                              <p
                                class="link text-muted"
                              >
                                Forgot Password?
                              </p>
                            </div>
                          </div>
                          <div
                            class="padding"
                          >
                            <button
                              class="button col-md-12 primary-button mt-5"
                              data-dismiss=""
                              data-target=""
                              data-testid="button"
                              data-toggle=""
                              id="submit"
                              type="submit"
                            >
                              Log In
                            </button>
                          </div>
                          <div
                            class="divider-box mt-2 mb-2"
                          >
                            <div
                              class="divider"
                            >
                              <span>
                                or
                              </span>
                            </div>
                          </div>
                          <div
                            class="padding"
                          >
                            <button
                              class="facebk-btn col-md-12"
                              data-dismiss=""
                              data-target=""
                              data-testid="button"
                              data-toggle=""
                              id="2jjaYY3qn"
                              type="button"
                            >
                              Facebook
                            </button>
                          </div>
                          <div
                            class="form-options padding"
                          >
                            <p>
                              New to Bukka?
                            </p>
                            <button
                              class="btn-link auth-footer-btn"
                              data-dismiss=""
                              data-target=""
                              data-testid="button"
                              data-toggle=""
                              id="/signup"
                              type="button"
                            >
                              SIGN UP
                            </button>
                          </div>
                        </form>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <nav
          class="container navbar navbar-light"
        >
          <div
            class="row mx-0"
          >
            <div>
              <a
                class="bukka-brand"
                href="/"
              >
                <div
                  class="bukka-logo"
                >
                  <img
                    alt=""
                    src="https://res.cloudinary.com/digitzs/image/upload/v1550480450/Bukka%20App/bukka-logo.svg"
                  />
                </div>
              </a>
            </div>
            <div
              class="pl-lg-5
            d-none d-md-none d-lg-inline-flex"
            >
              <div
                class="pr-17"
              >
                <div
                  class="position-relative"
                >
                  <div>
                    <button
                      class="current-location-button inherit-bg"
                      data-dismiss=""
                      data-target=""
                      data-testid="button"
                      data-toggle=""
                      id="2jjaYY3qn"
                      type="button"
                    >
                      <span
                        class="current-location-button-icon custom-mt-minus19"
                      >
                        <svg
                          class="magnifier"
                          height="16"
                          viewBox="0 0 16 16"
                          width="16"
                        >
                          <defs>
                            <path
                              d="M11.75 10.335l3.32 3.322-1.413 1.414-3.322-3.32A6.48 6.48 0 0 1 6.5 13a6.5 6.5 0 1 1 5.25-2.665zM6.5 11a4.5 4.5 0 1 0 0-9 4.5 4.5 0 0 0 0 9z"
                              id="icon-search_svg__a"
                            />
                          </defs>
                          <use
                            fill="currentColor"
                            fill-rule="evenodd"
                            opacity=".9"
                            xlink:href="#icon-search_svg__a"
                          />
                        </svg>
                      </span>
                      <div>
                        <div
                          class="current-location-button-text"
                        >
                          <input
                            accept=""
                            autocomplete="searchanything"
                            class="searchanything-text-field inherit-bg searchlocation"
                            name="searchLocation"
                            placeholder="Search for anything..."
                            type="text"
                            value=""
                          />
                        </div>
                      </div>
                    </button>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="form-inline"
          >
            <div
              class="d-none bukka-md-inline-flex"
            >
              <a
                class="bukka-btn"
                href="/feed"
              >
                Food
              </a>
              <a
                class="bukka-btn"
                href="/fresh"
              >
                Fresh
              </a>
              <a
                class="bukka-btn"
                href="/drinks"
              >
                Drinks
              </a>
            </div>
            <button
              class="small-outline-button bg-transparent"
              data-dismiss=""
              data-target="#authModal"
              data-testid="button"
              data-toggle="modal"
              id="/login"
              type="button"
            >
              sign in
            </button>
            <button
              class="small-button mr-0"
              data-dismiss=""
              data-target="#authModal"
              data-testid="button"
              data-toggle="modal"
              id="/signup"
              type="button"
            >
              sign up
            </button>
          </div>
        </nav>
      </div>
      <section
        class="feed-main-section pt-5"
      >
        <div
          class="mt-130 mb-4"
        >
          <div
            class="container "
          >
            <div
              class="headline"
            >
              <div
                class="runner"
                style="transform: translateX(100% - 32px);"
              />
              <h2
                class="headline-h2"
              >
                “ a ”
              </h2>
              <a
                class="headline-link"
                href="/"
                rel="nofollow"
              >
                <span
                  class="d-none pr-3 d-sm-inline-flex"
                >
                  6
                   
                  Results
                </span>
              </a>
            </div>
          </div>
          <div
            class="container px-0"
          >
            <div
              class="col-md-3 col-lg-2"
            >
              <div
                class="btn-group"
                role="group"
              >
                <button
                  class="btn-delivery delivery-mode-active"
                  type="button"
                >
                  delivery
                </button>
                <button
                  class="btn-pickup "
                  type="button"
                >
                  pickup
                </button>
              </div>
            </div>
          </div>
          <div
            class="container "
          >
            <div
              class="row pb-4"
            >
              <div
                class="card-container col-lg-4 col-md-4 col-sm-12"
              >
                <div
                  class="card-wrap"
                >
                  
                  <div
                    class="mt-4 bukka-card"
                    data-target="#bukkaAddToCart"
                    data-toggle="modal"
                  >
                    <img
                      alt="alt_image"
                      class="img-small-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595380/bukka-favorite-7.webp"
                    />
                    <img
                      alt="alt_image"
                      class="img-large-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595380/bukka-favorite-7.webp"
                    />
                    <div
                      class="details"
                    >
                      <h3>
                        <span
                          class="details-location"
                        >
                          miss saigon
                        </span>
                      </h3>
                      <div
                        class="delivery"
                      >
                        <h4
                          class="price-badge"
                        >
                          ₦
                          300
                          .00
                        </h4>
                        <span
                          class="custom-delivery-badge"
                        >
                          Delivery
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="card-container col-lg-4 col-md-4 col-sm-12"
              >
                <div
                  class="card-wrap"
                >
                  
                  <div
                    class="mt-4 bukka-card"
                    data-target="#bukkaAddToCart"
                    data-toggle="modal"
                  >
                    <img
                      alt="alt_image"
                      class="img-small-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595380/bukka-favorite-8.webp"
                    />
                    <img
                      alt="alt_image"
                      class="img-large-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595380/bukka-favorite-8.webp"
                    />
                    <div
                      class="details"
                    >
                      <h3>
                        <span
                          class="details-location"
                        >
                          chaat corner
                        </span>
                      </h3>
                      <div
                        class="delivery"
                      >
                        <h4
                          class="price-badge"
                        >
                          ₦
                          300
                          .00
                        </h4>
                        <span
                          class="custom-delivery-badge"
                        >
                          Delivery
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="card-container col-lg-4 col-md-4 col-sm-12"
              >
                <div
                  class="card-wrap"
                >
                  
                  <div
                    class="mt-4 bukka-card"
                    data-target="#bukkaAddToCart"
                    data-toggle="modal"
                  >
                    <img
                      alt="alt_image"
                      class="img-small-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595379/bukka-favorite-2.webp"
                    />
                    <img
                      alt="alt_image"
                      class="img-large-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595379/bukka-favorite-2.webp"
                    />
                    <div
                      class="details"
                    >
                      <h3>
                        <span
                          class="details-location"
                        >
                          jack in the box
                        </span>
                      </h3>
                      <div
                        class="delivery"
                      >
                        <h4
                          class="price-badge"
                        >
                          ₦
                          300
                          .00
                        </h4>
                        <span
                          class="custom-delivery-badge"
                        >
                          Delivery
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="card-container col-lg-4 col-md-4 col-sm-12"
              >
                <div
                  class="card-wrap"
                >
                  
                  <div
                    class="mt-4 bukka-card"
                    data-target="#bukkaAddToCart"
                    data-toggle="modal"
                  >
                    <img
                      alt="alt_image"
                      class="img-small-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595380/bukka-favorite-3.webp"
                    />
                    <img
                      alt="alt_image"
                      class="img-large-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595380/bukka-favorite-3.webp"
                    />
                    <div
                      class="details"
                    >
                      <h3>
                        <span
                          class="details-location"
                        >
                          ike's love and sandwishes
                        </span>
                      </h3>
                      <div
                        class="delivery"
                      >
                        <h4
                          class="price-badge"
                        >
                          ₦
                          300
                          .00
                        </h4>
                        <span
                          class="custom-delivery-badge"
                        >
                          Delivery
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="card-container col-lg-4 col-md-4 col-sm-12"
              >
                <div
                  class="card-wrap"
                >
                  
                  <div
                    class="mt-4 bukka-card"
                    data-target="#bukkaAddToCart"
                    data-toggle="modal"
                  >
                    <img
                      alt="alt_image"
                      class="img-small-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554595380/bukka-favorite-4.webp"
                    />
                    <img
                      alt="alt_image"
                      class="img-large-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554595380/bukka-favorite-4.webp"
                    />
                    <div
                      class="details"
                    >
                      <h3>
                        <span
                          class="details-location"
                        >
                          Urban pizza kitchen
                        </span>
                      </h3>
                      <div
                        class="delivery"
                      >
                        <h4
                          class="price-badge"
                        >
                          ₦
                          300
                          .00
                        </h4>
                        <span
                          class="custom-delivery-badge"
                        >
                          Delivery
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="card-container col-lg-4 col-md-4 col-sm-12"
              >
                <div
                  class="card-wrap"
                >
                  
                  <div
                    class="mt-4 bukka-card"
                    data-target="#bukkaAddToCart"
                    data-toggle="modal"
                  >
                    <img
                      alt="alt_image"
                      class="img-small-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1554495327/carousel-alcholic-drinks.webp"
                    />
                    <img
                      alt="alt_image"
                      class="img-large-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1554495327/carousel-alcholic-drinks.webp"
                    />
                    <div
                      class="details"
                    >
                      <h3>
                        <span
                          class="details-location"
                        >
                          alcholic drinks
                        </span>
                      </h3>
                      <div
                        class="delivery"
                      >
                        <h4
                          class="price-badge"
                        >
                          ₦
                          300
                          .00
                        </h4>
                        <span
                          class="custom-delivery-badge"
                        >
                          Delivery
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
